### PR TITLE
MediaElement Preload

### PR DESCRIPTION
**Pull request by X-Raym**
Add a ‘preload’ parameter to load function to choose the preload HTML5 audio attribute value if MediaElement.

### CODE REVIEW

File: dist/wavesurfer.js [hunk 1]
```diff
@@ -1192,14 +1192,15 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
...
      */
-    load: function (url, container, peaks) {
+    load: function (url, container, peaks, preload) {
         var my = this;
 
         var media = document.createElement(this.mediaType);
         media.controls = this.params.mediaControls;
         media.autoplay = this.params.autoplay || false;
-        media.preload = 'auto';
+		media.preload = preload == null ? 'auto' : preload;
```
Comment (katspaugh) [inline dist/wavesurfer.js]: The indentation is off.

File: src/mediaelement.js [hunk 2]
```diff
@@ -49,14 +49,15 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
      *  @param  {String}        url         path to media file
      *  @param  {HTMLElement}   container   HTML element
      *  @param  {Array}         peaks       array of peak data
+	 *  @param  {String}        preload     HTML 5 preload attribute value
```
Comment (katspaugh) [inline src/mediaelement.js]: Indtentation.

**Comment by katspaugh** [comment]
Looks good to me!

**Comment by X-Raym** [comment]
@katspaugh should be corrected now :P

File: dist/wavesurfer.js [hunk 3]
```diff
@@ -1192,14 +1192,15 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
...
      */
-    load: function (url, container, peaks) {
+    load: function (url, container, peaks, preload) {
         var my = this;
 
         var media = document.createElement(this.mediaType);
         media.controls = this.params.mediaControls;
         media.autoplay = this.params.autoplay || false;
-        media.preload = 'auto';
+	media.preload = preload == null ? 'auto' : preload;
```
Comment (katspaugh) [inline dist/wavesurfer.js]: Still 1 space wrong.

File: src/wavesurfer.js [hunk 4]
```diff
@@ -412,6 +412,7 @@ var WaveSurfer = {
                 this.decodeArrayBuffer(arraybuffer, (function (buffer) {
                     this.backend.buffer = buffer;
                     this.drawBuffer();
+                    this.fireEvent('waveform_ME_noPeaks');
```
Comment (katspaugh) [inline src/wavesurfer.js]: This is something extra, let's remove.

Comment (X-Raym) [inline src/wavesurfer.js]: Yep, it will be my next proposition, we'll talk about that later 😄

File: dist/wavesurfer.js (hunk 3 above)
Comment (X-Raym) [inline dist/wavesurfer.js]: Arf, github webditor shows this as perfectly aligned. I created the indentation by copy pasting from the above line.
Well, should be corrected now.

**Comment by X-Raym** [comment]
@katspaugh I hope indentations are solved now.

**Comment by katspaugh** [comment]
Merged, thanks!

**Comment by X-Raym** [comment]
@katspaugh Thanks for having accepted this request. Cheers !

File: dist/wavesurfer.js [hunk 5]
```diff
@@ -1192,14 +1192,15 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
      *  @param  {String}        url         path to media file
      *  @param  {HTMLElement}   container   HTML element
      *  @param  {Array}         peaks       array of peak data
+	 *  @param  {String}        preload     HTML 5 preload attribute value
```
Comment (katspaugh) [inline dist/wavesurfer.js]: Damn, the indentation is still wrong here.

Comment (X-Raym) [inline dist/wavesurfer.js]: Arf, GitHub web editor is really not friendly with indentation display... :/
Do you want me to fix this ou can you do it ?

Comment (katspaugh) [inline dist/wavesurfer.js]: I'll do it when doing a release, no worries.

Comment (X-Raym) [inline dist/wavesurfer.js]: @katspaugh Sorry for the inconvenience. Indentations are not displayed and created equally between editors. --'

**Comment by thijstriemstra** [comment]
@X-Raym can you also update the changelog?

**Comment by X-Raym** [comment]
@thijstriemstra ok, what version number should I put ?

**Comment by katspaugh** [comment]
1.2.3, @X-Raym

**Comment by X-Raym** [comment]
@thijstriemstra @katspaugh https://github.com/katspaugh/wavesurfer.js/pull/857 Done !

**Pull request merged**